File: leetsync/3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.java
class Solution {
    public int minOperations(int[] nums, int k) {
        PriorityQueue<Long> pq = new PriorityQueue<>();
        int minOperations = 0;
        for (int num : nums) {
            pq.add((long) num);
        }
        while (!pq.isEmpty() && (pq.size() > 1) && (pq.peek() < k)) {
            long first = pq.poll();
            long second = pq.poll();
            long third = Math.min(first, second) * 2L + Math.max(first, second);
            pq.add(third);
            minOperations++;
        }
        return minOperations;
    }
}

File: leetsync/3332-minimum-operations-to-exceed-threshold-value-ii/minimum-operations-to-exceed-threshold-value-ii.go
type MinHeap []int

func (h MinHeap) Len() int {
	return len(h)
}
func (h MinHeap) Less(i, j int) bool {
	return h[i] < h[j]
}
func (h MinHeap) Swap(i, j int) {
	h[i], h[j] = h[j], h[i]
}
func (h *MinHeap) Push(x interface{}) {
	*h = append(*h, x.(int))
}
func (h *MinHeap) Pop() interface{} {
	x := (*h)[len(*h)-1]
	*h = (*h)[:len(*h)-1]
	return x
}
func minOperations(nums []int, k int) int {
	h := &MinHeap{}
	heap.Init(h)
	for _, num := range nums {
		heap.Push(h, num)
	}
	var operations int
	for h.Len() > 1 && (*h)[0] < k {
		x := heap.Pop(h).(int)
		y := heap.Pop(h).(int)
		z := min(x, y)*2 + max(x, y)
		heap.Push(h, z)
		operations++
	}
	return operations
}